crates/transaction-pool/src/traits.rs

```diff
@@ -754,14 +754,19 @@ impl BestTransactionsAttributes {
 }
 
 /// Trait for transaction types used inside the pool
-pub trait PoolTransaction:
-    fmt::Debug
-    + Send
-    + Sync
-    + FromRecoveredPooledTransaction
-    + TryFromRecoveredTransaction
-    + IntoRecoveredTransaction
-{
+pub trait PoolTransaction: fmt::Debug + Send + Sync + Sized {
+    type Source: TryInto<Self> + TryFrom<TransactionSignedEcRecovered>;
```

Comment:
let's rename this to `Consensus`
```suggestion
    type Consensus: TryInto<Self> + TryFrom<TransactionSignedEcRecovered>;
```